rp2040-hal/src/i2c.rs

```diff
@@ -262,6 +286,19 @@ fn i2c_reserved_addr(addr: u16) -> bool {
     (addr & 0x78) == 0 || (addr & 0x78) == 0x78
 }
 
+fn configure_for_i2c<P: AnyPin>(pin: P) -> P {
+    // https://datasheets.raspberrypi.com/rp2040/rp2040-datasheet.pdf
+    // Page 441: 4.3.1.3. IOs
+    // • pull-up enabled
+    // • slew rate limited
```

Comment:
Well I'm not 100% sure but as I understand it, limiting the slew rate would also limit the maximum transfer speed, so it would be the "slow" mode. Therefore, `OutputSlewRate::Fast` seems to be the wrong choice.